numba/core/typed_passes.py

```diff
@@ -648,6 +651,42 @@ def _do_work_call(self, state, work_list, block, i, expr, inline_worker):
             work_list, is_method, inline_worker,
         )
 
+    def _do_work_ops(self, state, work_list, block, i, expr, inline_worker):
+
+        # search for this binop overloads with this "inline"
+        sig = state.type_annotation.calltypes[expr]
+        if sig is None:
+            return False
+
+        arg_typs = sig.args
+        templates = state.typingctx._functions.get(expr.fn, [])
+
+        impl = None
+        for template in templates:
+            inline_type = getattr(template, '_inline', None)
+            if inline_type is None:
+                # inline not defined
+                continue
+            if arg_typs not in template._inline_overloads:
+                # skip overloads not matching signature
+                continue
+            if not inline_type.is_never_inline:
+                try:
+                    impl = template._overload_func(*arg_typs)
+                    if impl is None:
+                        raise Exception  # abort for this template
+                    break
+                except Exception:
+                    continue
+        else:
+            return False
```

Comment:
This is a copy of the code above in `_do_work_call`? Perhaps it's time for a refactor?

Reply:
@stuartarchibald Sure, but after that please don't say it was better the old way. 😄 
I made some refactoring keeping the logic the same. Any comments and remarks are very welcome.
If that looks ugly in your opinion we can probably revert most of it and use only `select_template`  function to actually find the inlinee template in all three branches.